Library/Homebrew/cask/installer.rb

```diff
@@ -328,8 +328,9 @@ def collect_cask_and_formula_dependencies
 
     def missing_cask_and_formula_dependencies
       collect_cask_and_formula_dependencies.reject do |cask_or_formula|
-        (cask_or_formula.respond_to?(:installed?) && cask_or_formula.installed?) ||
-          (cask_or_formula.respond_to?(:any_version_installed?) && cask_or_formula.any_version_installed?)
+        ((cask_or_formula.respond_to?(:installed?) && cask_or_formula.installed?) ||
+          (cask_or_formula.respond_to?(:any_version_installed?) && cask_or_formula.any_version_installed?)) &&
+          (cask_or_formula.respond_to?(:linked?) && cask_or_formula.linked?)
```

Comment:
```suggestion
          (cask_or_formula.respond_to?(:linked?) && cask_or_formula.optlinked?)
```
otherwise this will always fail for keg-only formulae.

Also may want to use `try(:linked?)` from ActiveSupport to clean this up.